src/h3/mod.rs

```diff
@@ -1400,7 +1400,17 @@ impl Connection {
 
 /// Generates an HTTP/3 GREASE variable length integer.
 fn grease_value() -> u64 {
-    let n = std::cmp::min(super::rand::rand_u64(), 148_764_065_110_560_899);
+    let range = 148_764_065_110_560_899;
+    let chunk_size = u64::max_value() / range;
```

Comment:
This whole thing could be useful for other things, so should probably make a generic function in the `rand` submodule. Something like:
```
pub fn rand_u64_uniform(max: u64) -> u64
```